sdk/cmake/CMakeMacros.cmake

```diff
@@ -974,3 +974,13 @@ function(add_rostests_file)
         endif()
     endif()
 endfunction()
+
+if(PCH AND (NOT ENABLE_CCACHE) AND (NOT CMAKE_HOST_APPLE))
```

Comment:
Hmm it was done back then this way to allow anyone to experiment with both co-existing. I wouldn't see any reason to prevent that. What do you think?

Reply:
Does it make sense to use precompiled headers with ccache? From my understanding, it shouldn't increase performance in this case

Reply:
For me it doesn't make sense to mix them at all, but what I meant above is that I wouldn't restrict anyone from the possibility just because of that. It's not harming us in any way to leave that available.

Reply:
My point is simple: it's easier when pch build is enabled with only one cmake swith - `PCH`. If I want to turn it on or off, I just need to pass this switch manually and that's it. In this case, I need to also do something with `ENABLE_CCACHE` and `CMAKE_HOST_APPLE`

Reply:
My suggestion is to make
```
if(ENABLE_CCACHE OR CMAKE_HOST_APPLE)
set(PCH 0)
endif()
```

Reply:
I see what you mean. I moved the ccache check up around the pch check.